frontend/src/components/VaultDetails.vue

```diff
@@ -226,8 +226,7 @@ function permissionGranted() {
 }
 
 async function searchAuthority(query: string): Promise<AuthorityDto[]> {
-  return (await Promise.all([backend.users.search(query), backend.groups.search(query)]))
-    .flat()
+  return (await backend.authorities.search(query))
```

Comment:
Compared with the previous implementation this looses type information. Not a big deal in js, but still. Isn't the UI switching on the type in order to show different icons?

Reply:
Fixed in https://github.com/cryptomator/hub/pull/145/commits/b81b2fcbcabcac78104bd34dfa7428beb66b87d6 and already discussed it with @tobihagemann

Reply:
With https://github.com/cryptomator/hub/pull/145/commits/f2ebc7d5f0ae3ad2765701dd560f0daa3e2c2d52 all DTOs that are still classes are now instantiated with a type guard after validating the type, all others are types. In a later step the plan is to provide the other DTOs also with type-guards and to map these corresponding responses from the server as well (analogous to User/GroupDto).